packages/flutter_tools/lib/src/fuchsia/fuchsia_kernel_compiler.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../artifacts.dart';
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/logger.dart';
+import '../base/process_manager.dart';
+import '../build_info.dart';
+import '../convert.dart';
+import '../globals.dart';
+import '../project.dart';
+
+import 'fuchsia_sdk.dart';
+
+/// The [FuchsiaKernelCompiler] instance.
+FuchsiaKernelCompiler get fuchsiaKernelCompiler =>
+    context.get<FuchsiaKernelCompiler>();
+
+class FuchsiaKernelCompiler {
+  /// Compiles the [fuchsiaProject] with entrypoint [target] to a collection of
+  /// .dilp files, and a manifest that refers to them.
+  Future<void> build({
+    @required FuchsiaProject fuchsiaProject,
+    @required String target, // E.g., lib/main.dart
+    BuildInfo buildInfo = BuildInfo.debug,
+  }) async {
+    const String multiRootScheme = 'main-root';
```

Comment:
Will it be possible to make this configurable eventually? We also use a multi-root scheme to support the build_runner cache directory. Doesn't need to be handled in this PR.

Reply:
Added a TODO to pick up the filesystem roots and scheme from the BuildInfo.